compiler/src/dotty/tools/backend/sjs/JSDefinitions.scala

```diff
@@ -50,6 +52,11 @@ final class JSDefinitions()(using Context) {
   @threadUnsafe lazy val PseudoUnionType: TypeRef = requiredClassRef("scala.scalajs.js.|")
   def PseudoUnionClass(using Context) = PseudoUnionType.symbol.asClass
 
+  @threadUnsafe lazy val PseudoUnionModuleRef = requiredModuleRef("scala.scalajs.js.|")
```

Comment:
> Also note that we cannot "reinterpret" scala.scalajs.js.| as a dotty union type during Scala2Unpickler because they do not have the same erasure

The same is true for Scala 2 intersection types versus Dotty &, I have a wip branch which erases those differently, so we could extend that mechanism to handle unions too if needed.

Reply:
Yes that should be doable.